MailCore/Cache/MailboxManager/MailboxManager+Thread.swift

```diff
@@ -327,89 +322,82 @@ public extension MailboxManager {
     }
 
     private func deleteMessages(uids: [String]) async {
-        guard !uids.isEmpty && !Task.isCancelled else { return }
-
-        let backgroundTracker = await ApplicationBackgroundTaskTracker(identifier: #function + UUID().uuidString)
-        await backgroundRealm.execute { realm in
-            let batchSize = 100
-            for index in stride(from: 0, to: uids.count, by: batchSize) {
-                autoreleasepool {
-                    let uidsBatch = Array(uids[index ..< min(index + batchSize, uids.count)])
-
-                    let messagesToDelete = realm.objects(Message.self).where { $0.uid.in(uidsBatch) }
-                    var threadsToUpdate = Set<Thread>()
-                    var threadsToDelete = Set<Thread>()
-                    var draftsToDelete = Set<Draft>()
-
-                    for message in messagesToDelete {
-                        if let draft = self.draft(messageUid: message.uid, using: realm) {
-                            draftsToDelete.insert(draft)
-                        }
-                        for parent in message.threads {
-                            threadsToUpdate.insert(parent)
-                        }
+        guard !uids.isEmpty,
+              !Task.isCancelled else {
+            return
+        }
+
+        let batchSize = 100
+        for index in stride(from: 0, to: uids.count, by: batchSize) {
+            try? writeTransaction { writableRealm in
```

Comment:
Good ol switcheroo. 
The transaction is inside the for loop, and its implementation has an autoreleasepool. I'm expecting similar results memory wise as the previous solution.

Reply:
Before, `backgroundTracker` was preventing the app from suspending while executing the for loop. Now it could suspend in between for loop steps.

Reply:
Yeah I'll protect the entire thing just to be sure.

Reply:
done